src/utils/api.js

```diff
@@ -0,0 +1,147 @@
+// utils/Api.js
+
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+    // constructor body
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getUserInfo(), this.getInitialCards()])
+      .then(([userInfo, cards]) => {
+        // console.log("Cards:", cards); // Debugging
+        if (!Array.isArray(cards)) {
+          console.error("Expected an array but received:", cards);
+          return;
+        }
+
+        return [userInfo, cards];
+      })
+      .catch(console.error);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.